Dockerfile

```diff
@@ -1,2 +1,12 @@
+# Build
+FROM maven:3.8.4-openjdk-17 AS build-env
+WORKDIR /app
+COPY pom.xml ./
+RUN mvn dependency:go-offline
+COPY . ./
+RUN mvn package
```

Comment:
I think is better to use build scope and do it outside docker file, and create the docker image with a maven docker plugin.
Here in Docker file I'd just copy the resulting .jar.

Reply:
Good point, Including the build in the docker file, we make deploying easier, just needing a vm with docker. I think it not has impact from development perspective. 
Do you think we should change anyway??

Reply:
the FROM openjdk:17 is deleting previous one?
The final image should be the smallest and cleanest possible without build deps if this is accomplished we could have it here.
Other thing to have in mind, with maven you could trigger the whole release process, as deploy the artifact, create the docker image, push it, increase the version, etc.
I'm not sure this is the right place to do it, please let me know if I'm missing something

Reply:
I am not very convinced to have the docker image creation attached to maven build until we discuss where to push it. Perhaps a spike is required to clarify these definitions together?

Reply:
yeah it creates 2 different images one with the build and one only with the jar. We only need to update the pom version in each release as part of the release script. 
I think for the moment we can start with this approach.